src/libsyntax/parse/token.rs

```diff
@@ -201,6 +201,7 @@ impl Token {
             OpenDelim(..)               => true, // tuple, array or block
             Literal(..)                 => true, // literal
             Not                         => true, // operator not
+            Tilde                       => true, // operator not (incorrect)
```

Comment:
This will remove the suggestion from something like `return ~0`, but that's more or less ok.